tortoise/verifying.go

```diff
@@ -74,6 +75,26 @@ type verifying struct {
 	abstainedWeight map[types.LayerID]util.Weight
 	// total weight of good ballots from verified + 1 up to last processed
 	totalGoodWeight util.Weight
+	// layerReferenceHeight is a highest block height below reference height
+	layerReferenceHeight map[types.LayerID]uint64
+}
+
+func (v *verifying) onBlock(block *types.Block) {
+	if block.TickHeight > v.referenceHeight[block.LayerIndex.GetEpoch()] {
```

Comment:
this caused the node not to vote for a block higher than the medium epoch height, right?

Reply:
this method is called on every block so that we can find the max block height before reference epoch height

not sure how it is related to voting